tests/phpunit/tests/script-modules/wpScriptModules.php

```diff
@@ -126,6 +126,33 @@ public function test_wp_dequeue_script_module() {
 		$this->assertTrue( isset( $enqueued_script_modules['bar'] ) );
 	}
 
+
+	/**
+	* Tests that a script module can be deregistered
+	* after being enqueued, and that will be removed
+	* from the enqueue list too.
+	*
+	* @ticket 60463
+	*
+	* @covers ::register()
+	* @covers ::enqueue()
+	* @covers ::deregister()
+	* @covers ::print_enqueued_script_modules()
+	*/
+	public function test_wp_deregister_script_module() {
+		$this->script_modules->register( 'foo', '/foo.js' );
+		$this->script_modules->register( 'bar', '/bar.js' );
+		$this->script_modules->enqueue( 'foo' );
+		$this->script_modules->enqueue( 'bar' );
+		$this->script_modules->deregister( 'foo' ); // Dequeued.
+
+		$enqueued_script_modules = $this->get_enqueued_script_modules();
+
+		$this->assertCount( 1, $enqueued_script_modules );
+		$this->assertFalse( isset( $enqueued_script_modules['foo'] ) );
+		$this->assertTrue( isset( $enqueued_script_modules['bar'] ) );
+	}
+
```

Comment:
It would be nice to include also a test for the case when someone tries to deregister a module already deregistered by someone else or never registered.

Reply:
Done!